src/api/port.js

```diff
@@ -0,0 +1,242 @@
+import curry from 'lodash/curry';
+import flow from 'lodash/flow';
+import Immutable from 'immutable';
+import { PortRecord, PositionRecord } from '../constants/flowdesigner.model';
+import { PORT_SOURCE, PORT_SINK } from '../constants/flowdesigner.constants';
+
+const positionSelector = ['graphicalAttributes', 'position'];
+const componentTypeSelector = ['graphicalAttributes', 'portType'];
+const portTopologySelector = ['graphicalAttributes', 'properties', 'type'];
+const indexSelector = ['graphicalAttributes', 'properties', 'index'];
+
+export function isPositionRecord(position, doThrow) {
```

Comment:
can you provide a default value to "doThrow" like you did in the next function "isPortRecord"?